stl/inc/vector

```diff
@@ -1761,15 +1704,40 @@ private:
         _My_data._Orphan_all();
 
         if (_Myfirst) { // destroy and deallocate old array
-            _Destroy(_Myfirst, _Mylast);
-            _Getal().deallocate(_Myfirst, static_cast<size_type>(_Myend - _Myfirst));
+            _Destroy_range(_Myfirst, _Mylast, _Al);
+            _Al.deallocate(_Myfirst, static_cast<size_type>(_Myend - _Myfirst));
 
             _Myfirst = nullptr;
             _Mylast  = nullptr;
             _Myend   = nullptr;
         }
     }
 
+    template <class... _Valty>
+    _CONSTEXPR20_CONTAINER void _Construct_n(_CRT_GUARDOVERFLOW const size_type _Count, _Valty&&... _Val) {
+        auto& _Al       = _Getal();
+        auto&& _Alproxy = _GET_PROXY_ALLOCATOR(_Alty, _Al);
+        auto& _My_data  = _Mypair._Myval2;
+        _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _My_data);
+        if (_Count != 0) {
+            _Buy_nonzero(_Count);
+            _Tidy_guard<vector> _Guard{this};
+            if constexpr (sizeof...(_Val) == 0) {
+                _My_data._Mylast = _Uninitialized_value_construct_n(_My_data._Myfirst, _Count, _Al);
+            } else if constexpr (sizeof...(_Val) == 1) {
+                static_assert(is_same_v<_Valty..., const _Ty&>, "Wrong type passed to _Construct_n");
+                _My_data._Mylast = _Uninitialized_fill_n(_My_data._Myfirst, _Count, _Val..., _Al);
+            } else if constexpr (sizeof...(_Val) == 2) {
+                _My_data._Mylast = _Uninitialized_copy(_STD forward<_Valty>(_Val)..., _My_data._Myfirst, _Al);

```

Comment:
@CaseyCarter, I hope you will appreciate that we now perfectly forward user defined iterators

Can you feel the performance?